src/components/cards/PairPriceGraph/PairPriceGraph.vue

```diff
@@ -0,0 +1,323 @@
+import { coingeckoService } from '@/services/coingecko/coingecko.service';
+import { useQuery } from 'vue-query';
+import {
+  Dictionary,
+  mapKeys,
+  mapValues,
+  maxBy,
+  minBy,
+  pickBy,
+  toPairs
+} from 'lodash';
+import { fromUnixTime, format } from 'date-fns';
+import useTailwind from '@/composables/useTailwind';
+import useBreakpoints from '@/composables/useBreakpoints';
+import { useTradeState } from '@/composables/trade/useTradeState';
+import { getAddress } from '@ethersproject/address';
+const easing = 'spring(1, 150, 18, 0)';
+
+async function getPairPriceData(
+  inputAsset: string,
+  outputAsset: string,
+  days: number
+) {
+  const aggregateBy = days === 1 ? 'hour' : 'day';
+  const inputAssetData = await coingeckoService.prices.getTokensHistorical(
+    [inputAsset],
+    days,
+    1,
+    aggregateBy
+  );
+
+  const outputAssetData = await coingeckoService.prices.getTokensHistorical(
+    [outputAsset],
+    days,
+    1,
+    aggregateBy
+  );
+
+  const calculatedPricing = mapValues(inputAssetData, (value, timestamp) => {
+    if (!outputAssetData[timestamp]) return null;
+    return (1 / value[0]) * outputAssetData[timestamp][0];
+  });
+
+  const calculatedPricingNoNulls = pickBy(calculatedPricing) as Dictionary<
+    number
+  >;
+
+  const formatTimestamps = mapKeys(
+    calculatedPricingNoNulls,
+    (_, timestamp: any) =>
+      format(fromUnixTime(timestamp / 1000), 'yyyy/MM/dd HH:mm')
+  );
+  return toPairs(formatTimestamps);
+}
+
+const chartTimespans = [
+  {
+    option: '1d',
+    value: 1
+  },
+  {
+    option: '1w',
+    value: 7
+  },
+  {
+    option: '1m',
+    value: 30
+  },
+  {
+    option: '1y',
+    value: 365
+  },
+  {
+    option: 'All',
+    value: -1
+  }
+];
+
+const { upToLargeBreakpoint } = useBreakpoints();
+const store = useStore();
+const { tokens } = useTokens();
+const { tokenInAddress, tokenOutAddress } = useTradeState();
+const tailwind = useTailwind();
+
+const animateInstance = ref();
+const elementToAnimate = ref<HTMLElement>();
+const chartHeight = ref(upToLargeBreakpoint ? 75 : 100);
+const activeTimespan = ref(chartTimespans[0]);
+const isExpanded = ref(false);
+const resizeTick = ref(0);
+
+const appLoading = computed(() => store.state.app.loading);
+
+const inputSym = computed(
+  () => tokens.value[getAddress(tokenInAddress.value)]?.symbol
+);
+const outputSym = computed(
+  () => tokens.value[getAddress(tokenOutAddress.value)]?.symbol
+);
+
+const shouldLoadPriceData = computed(
+  () => tokenInAddress.value !== '' && tokenOutAddress.value !== ''
+);
+
+const dataMin = computed(() => {
+  return (minBy(priceData.value || [], v => v[1]) || [])[1] || 0;
+});
+
+const dataMax = computed(() => {
+  return (maxBy(priceData.value || [], v => v[1]) || [])[1] || 0;
+});
+
+const {
+  isLoading: isLoadingPriceData,
+  data: priceData,
+  error: failedToLoadPriceData
+} = useQuery(
+  reactive([
+    'pairPriceData',
+    { tokenInAddress, tokenOutAddress, activeTimespan }
```

Comment:
could you also add `networkId` here? (also same comment about moving it to `queryKeys.ts`)

Reply:
ye